sdk/python/kfp/components/_data_passing.py

```diff
@@ -139,23 +139,24 @@ def get_canonical_type_for_type_struct(type_struct) -> str:
         return None
 
 
-def get_deserializer_code_for_type(typ) -> str:
-    try:
-        return type_name_to_deserializer.get(get_canonical_type_struct_for_type[typ], None)
-    except:
-        return None
+# Doesn't seem to be used anymore, can be removed?
+# def get_deserializer_code_for_type(typ) -> str:
+#     try:
+#         return type_name_to_deserializer.get(get_canonical_type_struct_for_type[typ], None)
+#     except:
+#         return None
 
 
-def get_deserializer_code_for_type_struct(type_struct) -> str:
+def get_deserializer_code_for_type_struct(type_struct: str) -> str:
     try:
-        return type_name_to_deserializer.get(type_struct, None)
+        return type_name_to_deserializer.get(type_annotation_utils.get_short_type_name(type_struct), None)
     except:
         return None
```

Comment:
Hmm, will this `except` clause ever be reached? In the `try` class, you do a `get` and provide a default `None`, so it should never raise an exception?

Reply:
Yeah good question, it get's triggered when `type_struct` is not a `str`. Not ideal as ideally the caller of these methods are conforming to the designated `type` of the input arguments but to be honest I didn't fancy trying to fix all the references to this across `kfp` so I just kept the `except` in place 😄

Reply:
Ahh makes sense, thanks for letting me know!